app/models/firmware_registry_simple.rb

```diff
@@ -0,0 +1,59 @@
+class FirmwareRegistrySimple < FirmwareRegistry
+  def sync_fw_binaries_raw
+    remote_binaries.each do |binary_hash|
+      binary = FirmwareBinary.find_or_create_by(:firmware_registry => self, :external_ref => binary_hash['id'])
+      _log.info("Updating FirmwareBinary [#{binary.id} | #{binary.name}]...")
+
+      binary.name = binary_hash['filename'] || binary_hash['id']
+      binary.description = binary_hash['description']
+      binary.version = binary_hash['version']
+      binary.save!
+
+      unless binary.urls.sort == binary_hash['urls'].sort
+        _log.info("Updating FirmwareBinary [#{binary.id} | #{binary.name}] endpoints...")
+        binary.endpoints.destroy_all
+        binary.endpoints = binary_hash['urls'].map { |url| Endpoint.create(:url => url) }
+      end
+
+      currents = binary.firmware_targets.map(&:to_hash)
+      remotes = binary_hash['compatible_server_models'].map { |r| r.symbolize_keys.slice(*FirmwareTarget::MATCH_ATTRIBUTES) }
+      unless currents.map(&:to_a).sort == remotes.map(&:to_a).sort
```

Comment:
Do you have an example of what the data here would look like?  Having a hard time picturing what is going on with the `.map(&:to_hash)` above then the `.map(&:to_a)` here

Reply:
I do https://github.com/ManageIQ/manageiq/pull/18774/files?file-filters%5B%5D=.json&owned-by%5B%5D=#diff-98d768062372d6463bc80189df295c18